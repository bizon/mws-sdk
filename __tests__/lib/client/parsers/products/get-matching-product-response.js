const parseXml = require('../../../../../lib/client/parsers')
const parseGetMatchingProductResponse = require('../../../../../lib/client/parsers/products/get-matching-product-response')

describe('lib.client.parsers.products.get-matching-product-response', () => {
  it('should parse the GetMatchingProductResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetMatchingProductResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <GetMatchingProductResult ASIN="B002KT3XRQ" status="Success">
          <Product xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01"
            xmlns:ns2="http://mws.amazonservices.com/schema/Products/2011-10-01/default.xsd">
            <Identifiers>
              <MarketplaceASIN>
                <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                <ASIN>B002KT3XRQ</ASIN>
              </MarketplaceASIN>
            </Identifiers>
            <AttributeSets>
              <ns2:ItemAttributes xml:lang="en-US">
                <ns2:Binding>Apparel</ns2:Binding>
                <ns2:Brand>Pearl iZUMi</ns2:Brand>
                <ns2:Department>mens</ns2:Department>
                <ns2:Feature>
                  Select transfer fabric sets the benchmark for moisture transfer and
                  four-way performance stretch
                </ns2:Feature>
                <ns2:Feature>
                  6-Panel anatomic design for superior, chafe-free comfort, UPF 50+ sun
                  protection
                </ns2:Feature>
                <ns2:Feature>
                  Comfortable silicone leg grippers keep shorts in place,
                  moisture-wicking, antimicrobial properties
                </ns2:Feature>
                <ns2:Feature>
                  Tour 3D Chamois is male-specific and non-chafing with padding in
                  key areas
                </ns2:Feature>
                <ns2:Feature>86 percent nylon, 14% spandex, 9-Inch inseam</ns2:Feature>
                <ns2:ItemDimensions>
                  <ns2:Height Units="inches">2.00</ns2:Height>
                  <ns2:Length Units="inches">9.00</ns2:Length>
                  <ns2:Width Units="inches">9.00</ns2:Width>
                </ns2:ItemDimensions>
                <ns2:Label>Pearl iZUMi</ns2:Label>
                <ns2:ListPrice>
                  <ns2:Amount>50.00</ns2:Amount>
                  <ns2:CurrencyCode>USD</ns2:CurrencyCode>
                </ns2:ListPrice>
                <ns2:Manufacturer>Pearl iZUMi</ns2:Manufacturer>
                <ns2:Model>0275</ns2:Model>
                <ns2:PackageDimensions>
                  <ns2:Height Units="inches">2.80</ns2:Height>
                  <ns2:Length Units="inches">9.75</ns2:Length>
                  <ns2:Width Units="inches">8.40</ns2:Width>
                  <ns2:Weight Units="pounds">0.40</ns2:Weight>
                </ns2:PackageDimensions>
                <ns2:PackageQuantity>1</ns2:PackageQuantity>
                <ns2:PartNumber>0275</ns2:PartNumber>
                <ns2:ProductGroup>Apparel</ns2:ProductGroup>
                <ns2:ProductTypeName>SHORTS</ns2:ProductTypeName>
                <ns2:Publisher>Pearl iZUMi</ns2:Publisher>
                <ns2:SmallImage>
                  <ns2:URL>
                    http://ecx.images-amazon.com/images/I/41ty3Sn%2BU8L._SL75_.jpg
                  </ns2:URL>
                  <ns2:Height Units="pixels">75</ns2:Height>
                  <ns2:Width Units="pixels">58</ns2:Width>
                </ns2:SmallImage>
                <ns2:Studio>Pearl iZUMi</ns2:Studio>
                <ns2:Title>Pearl iZUMi Men's Quest Cycling Short</ns2:Title>
              </ns2:ItemAttributes>
            </AttributeSets>
            <Relationships>
              <ns2:VariationChild>
                <Identifiers>
                  <MarketplaceASIN>
                    <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                    <ASIN>B002KT3XQC</ASIN>
                  </MarketplaceASIN>
                </Identifiers>
                <ns2:Color>Black</ns2:Color>
                <ns2:Size>Small</ns2:Size>
              </ns2:VariationChild>
              <ns2:VariationChild>
                <Identifiers>
                  <MarketplaceASIN>
                    <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                    <ASIN>B002KT3XQW</ASIN>
                  </MarketplaceASIN>
                </Identifiers>
                <ns2:Color>Black</ns2:Color>
                <ns2:Size>Medium</ns2:Size>
              </ns2:VariationChild>
              <ns2:VariationChild>
                <Identifiers>
                  <MarketplaceASIN>
                    <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                    <ASIN>B002KT3XQM</ASIN>
                  </MarketplaceASIN>
                </Identifiers>
                <ns2:Color>Black</ns2:Color>
                <ns2:Size>Large</ns2:Size>
              </ns2:VariationChild>
              <ns2:VariationChild>
                <Identifiers>
                  <MarketplaceASIN>
                    <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                    <ASIN>B002KT3XR6</ASIN>
                  </MarketplaceASIN>
                </Identifiers>
                <ns2:Color>Black</ns2:Color>
                <ns2:Size>X-Large</ns2:Size>
              </ns2:VariationChild>
              <ns2:VariationChild>
                <Identifiers>
                  <MarketplaceASIN>
                    <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                    <ASIN>B002KT3XRG</ASIN>
                  </MarketplaceASIN>
                </Identifiers>
                <ns2:Color>Black</ns2:Color>
                <ns2:Size>XX-Large</ns2:Size>
              </ns2:VariationChild>
            </Relationships>
            <SalesRankings>
              <SalesRank>
                <ProductCategoryId>apparel_display_on_website</ProductCategoryId>
                <Rank>159</Rank>
              </SalesRank>
              <SalesRank>
                <ProductCategoryId>2420095011</ProductCategoryId>
                <Rank>1</Rank>
              </SalesRank>
              <SalesRank>
                <ProductCategoryId>2611189011</ProductCategoryId>
                <Rank>4</Rank>
              </SalesRank>
            </SalesRankings>
          </Product>
        </GetMatchingProductResult>
        <ResponseMetadata>
          <RequestId>b12caada-d330-4d87-a789-EXAMPLE35872</RequestId>
        </ResponseMetadata>
      </GetMatchingProductResponse>`,
    )

    const response = parseGetMatchingProductResponse('/products:GetMatchingProductResponse', doc)

    expect(response).toMatchSnapshot()
  })
})
