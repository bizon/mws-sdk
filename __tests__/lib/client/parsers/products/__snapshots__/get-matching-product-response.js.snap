// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.products.get-matching-product-response should parse the GetMatchingProductResponse example response from MWS doc 1`] = `
Object {
  "getMatchingProductResults": Array [
    Object {
      "asin": "B002KT3XRQ",
      "product": Object {
        "attributes": Object {
          "binding": "Apparel",
          "brand": "Pearl iZUMi",
          "department": "mens",
          "features": Array [
            "Select transfer fabric sets the benchmark for moisture transfer and
                  four-way performance stretch",
            "6-Panel anatomic design for superior, chafe-free comfort, UPF 50+ sun
                  protection",
            "Comfortable silicone leg grippers keep shorts in place,
                  moisture-wicking, antimicrobial properties",
            "Tour 3D Chamois is male-specific and non-chafing with padding in
                  key areas",
            "86 percent nylon, 14% spandex, 9-Inch inseam",
          ],
          "itemDimensions": Object {
            "height": Object {
              "unit": "inches",
              "value": 2,
            },
            "length": Object {
              "unit": "inches",
              "value": 9,
            },
            "weight": null,
            "width": Object {
              "unit": "inches",
              "value": 9,
            },
          },
          "label": "Pearl iZUMi",
          "listPrice": Object {
            "amount": 50,
            "currencyCode": "USD",
          },
          "manufacturer": "Pearl iZUMi",
          "model": "0275",
          "packageDimensions": Object {
            "height": Object {
              "unit": "inches",
              "value": 2.8,
            },
            "length": Object {
              "unit": "inches",
              "value": 9.75,
            },
            "weight": Object {
              "unit": "pounds",
              "value": 0.4,
            },
            "width": Object {
              "unit": "inches",
              "value": 8.4,
            },
          },
          "packageQuantity": 1,
          "partNumber": "0275",
          "productGroup": "Apparel",
          "productTypeName": "SHORTS",
          "publisher": "Pearl iZUMi",
          "smallImage": Object {
            "height": Object {
              "unit": "pixels",
              "value": 75,
            },
            "url": "http://ecx.images-amazon.com/images/I/41ty3Sn%2BU8L._SL75_.jpg",
            "width": Object {
              "unit": "pixels",
              "value": 58,
            },
          },
          "studio": "Pearl iZUMi",
          "title": "Pearl iZUMi Men's Quest Cycling Short",
        },
        "identifiers": Object {
          "marketplaceAsin": Object {
            "asin": "B002KT3XRQ",
            "marketplaceId": "ATVPDKIKX0DER",
          },
          "skuIdentifier": null,
        },
        "offers": Array [],
        "relationships": Object {
          "variationChildren": Array [
            Object {
              "color": "Black",
              "identifiers": Object {
                "marketplaceAsin": Object {
                  "asin": "B002KT3XQC",
                  "marketplaceId": "ATVPDKIKX0DER",
                },
                "skuIdentifier": null,
              },
              "size": "Small",
            },
            Object {
              "color": "Black",
              "identifiers": Object {
                "marketplaceAsin": Object {
                  "asin": "B002KT3XQW",
                  "marketplaceId": "ATVPDKIKX0DER",
                },
                "skuIdentifier": null,
              },
              "size": "Medium",
            },
            Object {
              "color": "Black",
              "identifiers": Object {
                "marketplaceAsin": Object {
                  "asin": "B002KT3XQM",
                  "marketplaceId": "ATVPDKIKX0DER",
                },
                "skuIdentifier": null,
              },
              "size": "Large",
            },
            Object {
              "color": "Black",
              "identifiers": Object {
                "marketplaceAsin": Object {
                  "asin": "B002KT3XR6",
                  "marketplaceId": "ATVPDKIKX0DER",
                },
                "skuIdentifier": null,
              },
              "size": "X-Large",
            },
            Object {
              "color": "Black",
              "identifiers": Object {
                "marketplaceAsin": Object {
                  "asin": "B002KT3XRG",
                  "marketplaceId": "ATVPDKIKX0DER",
                },
                "skuIdentifier": null,
              },
              "size": "XX-Large",
            },
          ],
          "variationParent": null,
        },
        "salesRankings": Array [
          Object {
            "productCategoryId": "apparel_display_on_website",
            "rank": 159,
          },
          Object {
            "productCategoryId": "2420095011",
            "rank": 1,
          },
          Object {
            "productCategoryId": "2611189011",
            "rank": 4,
          },
        ],
      },
      "status": "Success",
    },
  ],
  "responseMetadata": Object {
    "requestId": "b12caada-d330-4d87-a789-EXAMPLE35872",
  },
}
`;
